toolkit/themes/shared/design-system/config.js

```diff
@@ -50,7 +52,7 @@ const MEDIA_QUERY_PROPERTY_MAP = {
 function hcmFormatter(args) {
   return (
     customFileHeader() +
-    ":root {\n" +
+    BASE_SELECTOR + "{\n" +
```

Comment:
This may format more nicely with the template string, could potentially avoid the const too

Reply:
I was trying template strings originally but it was kind of a pain as I kept getting weird indentations in the built CSS so would have to do something like this (which was still adding extra newline chars...):

<img width="473" alt="Screenshot 2024-02-06 at 9 45 42 AM" src="https://github.com/FirefoxUX/gecko-dev/assets/15138046/9643fc6c-7ae3-4cdb-a674-5c9493ca0e98">

The fact that we don't totally control the output of `formattedVariables` means it's easy to end up with this kind of spacing:

<img width="280" alt="Screenshot 2024-02-06 at 9 47 35 AM" src="https://github.com/FirefoxUX/gecko-dev/assets/15138046/819ec416-4a5c-4ee5-9957-d018f86de0cb">

So I dunno, we're already mostly doing string concatenation in the file, figured I would just stick with that rather than wrestle with unexpected spacing 🤷🏻

Reply:
Weird, I tried converting it to a template string and it seemed to work

<details>
<summary>Example patch</summary>

```diff
diff --git a/toolkit/themes/shared/design-system/config.js b/toolkit/themes/shared/design-system/config.js
index cdd2dbeead76..5ca9820b1496 100644
--- a/toolkit/themes/shared/design-system/config.js
+++ b/toolkit/themes/shared/design-system/config.js
@@ -7,37 +7,26 @@
 const StyleDictionary = require("style-dictionary");
 const { formattedVariables } = StyleDictionary.formatHelpers;
 
-/**
- * Adds the Mozilla Public License header in one comment and
- * how to make changes in the generated output files via the
- * design-tokens.json file in another comment.
- *
- * @return {String} Formatted comment header string
- */
-let customFileHeader = () => {
-  let licenseString = [
-    "/**",
-    " * This Source Code Form is subject to the terms of the Mozilla Public",
-    " * License, v. 2.0. If a copy of the MPL was not distributed with this",
-    " * file, You can obtain one at http://mozilla.org/MPL/2.0/.",
-    " */",
-  ].join("\n");
-
-  let commentString = [
-    "/**",
-    " * Do not edit this file directly, instead modify design-tokens.json",
-    " * and run `npm run build` to see your changes.",
-    " */",
-  ].join("\n");
-  return [licenseString + "\n\n" + commentString + "\n\n"];
-};
+const LICENSE_STRING = [
+  "/**",
+  " * This Source Code Form is subject to the terms of the Mozilla Public",
+  " * License, v. 2.0. If a copy of the MPL was not distributed with this",
+  " * file, You can obtain one at http://mozilla.org/MPL/2.0/.",
+  " */",
+].join("\n");
+
+const COMMENT_STRING = [
+  "/**",
+  " * Do not edit this file directly, instead modify design-tokens.json",
+  " * and run `npm run build` to see your changes.",
+  " */",
+].join("\n");
 
 const MEDIA_QUERY_PROPERTY_MAP = {
   "forced-colors": "forcedColors",
   "prefers-contrast": "prefersContrast",
 };
 
-const BASE_SELECTOR = ":root,\n" + ":host(.anonymous-content-host) {\n";
 
 /**
  * Formats built CSS to include "prefers-contrast" and "forced-colors" media
@@ -50,14 +39,19 @@ const BASE_SELECTOR = ":root,\n" + ":host(.anonymous-content-host) {\n";
  *  Formatted CSS including media queries.
  */
 function hcmFormatter(args) {
-  return (
-    customFileHeader() +
-    BASE_SELECTOR +
-    formatTokens({ args }) +
-    formatTokens({ mediaQuery: "prefers-contrast", args }) +
-    formatTokens({ mediaQuery: "forced-colors", args }) +
-    "}\n"
-  ).replaceAll(/(?<tokenName>\w+)-base(?=\b)/g, "$<tokenName>");
+  return `${LICENSE_STRING}
+
+${COMMENT_STRING}
+
+:root,
+:host(.anonymous-content-host) {
+${formatTokens({ args })}
+
+${formatTokens({ mediaQuery: "prefers-contrast", args })}
+
+${formatTokens({ mediaQuery: "forced-colors", args })}
+}
+`.replaceAll(/(?<tokenName>\w+)-base(?=\b)/g, "$<tokenName>");
 }
 
 /**
@@ -99,13 +93,11 @@ function formatTokens({ mediaQuery, args }) {
 
   // Weird spacing below is unfortunately necessary formatting the built CSS.
   if (mediaQuery) {
-    return `
-  @media (${mediaQuery}) {
+    return `  @media (${mediaQuery}) {
 ${formattedVars}
+  }`;
   }
-`;
-  }
-  return formattedVars + "\n";
+  return formattedVars;
 }
 
 /**
```

</details>

Reply:
Wasn't expecting that to fully post 😅 this is fine as is too